### PR TITLE
Fix test suites

### PR DESCRIPTION
**Pull request by klntsky**
### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: test/Unit.purs [hunk 1]
```diff
@@ -38,11 +42,15 @@ import Test.Ctl.Types.TokenName as Types.TokenName
...
 
 -- Run with `spago test --main Test.Ctl.Unit`
 main :: Effect Unit
-main = launchAff_ do
-  interpret testPlan
+main = interruptOnSignal SIGINT =<< launchAff do
+  flip cancelWith (effectCanceler (exitCode 1)) do
+    interpretWithConfig
+      defaultConfig { timeout = Just $ Milliseconds 30_000.0, exit = true }
+      testPlan
```
Comment (jy14898) [inline test/Unit.purs:53]: Probably not necessary, seeing as we don't launch plutip in unit/integration. However, it doesn't hurt to have this (enables graceful shutdown of Aff)

**Review by jy14898 — APPROVED** [review]

File: test/Integration.purs [hunk 2]
```diff
@@ -20,11 +24,15 @@ import Test.Ctl.BalanceTx.Time as BalanceTx.Time
...
+import Test.Spec.Runner (defaultConfig)
 
 -- Run with `spago test --main Test.Ctl.Integration`
 main :: Effect Unit
-main = launchAff_ do
-  interpret testPlan
+main = interruptOnSignal SIGINT =<< launchAff do
+  flip cancelWith (effectCanceler (exitCode 1)) do
+    interpretWithConfig
+      defaultConfig { timeout = Just $ Milliseconds 450_000.0, exit = true }
```
Comment (uhbif19) [inline test/Integration.purs:34]: What does those numbers mean?

File: test/Unit.purs [hunk 3]
```diff
@@ -38,11 +42,15 @@ import Test.Ctl.Types.TokenName as Types.TokenName
...
 import Test.Ctl.Wallet.Cip30.SignData as Cip30SignData
+import Test.Spec.Runner (defaultConfig)
 
 -- Run with `spago test --main Test.Ctl.Unit`
 main :: Effect Unit
-main = launchAff_ do
-  interpret testPlan
+main = interruptOnSignal SIGINT =<< launchAff do
+  flip cancelWith (effectCanceler (exitCode 1)) do
+    interpretWithConfig
```
Comment (uhbif19) [inline test/Unit.purs:51]: There is usages of interpret in some other small modules, like MustSpendTotal. Should they be changed too?

Comment (uhbif19) [reply test/Unit.purs:51]: Or this is required only for CI-related stuff?

Comment (klntsky) [reply test/Unit.purs:51]: No, these `main`s are not parts of our test suite (could be removed perhaps). But the MustSpendTotal suite itself is a part of unit tests

**Pull request merged**